src/gpu/intel/gemm/jit/generator/pieces/stream_k.cxx

```diff
@@ -128,8 +128,8 @@ void Generator<hw>::gemmStreamKSetup(Label &lKVPhaseDone, Label &lKernelDone,
         // Each workgroup gets a roughly k0-sized range in k space, which may span
         //  multiple C tiles.
 
-        // Traverse tiles backward:
-        eadd3(1, temp, nonKSyncedWGs, -slicedTileIdx, -1);
+        // Traverse tiles:
+        add(1, temp , slicedTileIdx, 1);
```

Comment:
@dyoussif Traversing tiles backward is important for performance, as it reduces/removes delays due to inter-WG synchronization. What was the exact case causing problems? (i.e. values of kUnsynced, kSlicedTiles, etc.)